### DISCUSSION TITLE
use skywalking to monitor apisix

### DISCUSSION THREAD

**Discussion by zjc1750514326** [apache/skywalking#12674] [Q&A] [answered]
I had see the document.
https://skywalking.apache.org/docs/main/v9.7.0/en/setup/backend/backend-apisix-monitoring/#apisix-monitoring
I do my apisix prometheus plugin,and `curl http://127.0.0.1:9091/apisix/prometheus/metrics` get data.
and i install OpenTelemetry Collector. it work. but i do not understand:
 OpenTelemetry Collector fetches metrics from [APISIX Prometheus plugin](https://apisix.apache.org/docs/apisix/plugins/prometheus/) via Prometheus Receiver and pushes metrics to SkyWalking OAP Server via OpenTelemetry gRPC exporter.
how to make data to Prometheus Receiver  and to skywalking?
this config will use where?
https://github.com/apache/skywalking/blob/690563ee92aac2146a3860425fd6e2817c8e52dc/test/e2e-v2/cases/apisix/otel-collector/otel-collector-config.yaml

**Comment by wu-sheng** [comment]
Yes, this is. We have this and showcase(repo) to demonstrate how to set this up.

**Reply by wu-sheng** [reply]
> how to make data to Prometheus Receiver and to skywalking?

OTEL collector read from apisix Prometheus endpoint.

**Reply by zjc1750514326** [reply, marked as answer]
well, it means:
first i should use OTEL collector  set config like this 
https://github.com/apache/skywalking/blob/1ca62dcbaa60f6fc87cd2230abb438071dd5803f/test/e2e-v2/cases/apisix/otel-collector/otel-collector-config.yaml
make sure OTEL collector get data.
second i should use OTEL collector tranlate data to skywalking?

**Reply by wu-sheng** [reply]
Yes. It is.

**Reply by zjc1750514326** [reply]
oh thank, i will try this solution.

**Reply by wu-sheng** [reply]
showcase(https://skywalking.apache.org/docs/skywalking-showcase/next/readme/) provided a demo setup. 
If you know k8s, that is a super easy demo.

**Reply by zjc1750514326** [reply]
> showcase(https://skywalking.apache.org/docs/skywalking-showcase/next/readme/) provided a demo setup. If you know k8s, that is a super easy demo.

i do not konw k8s, i had do samething:
1: i have make sure apisix plugin working.request url get data.
2: i has install otelcol and change config https://github.com/apache/skywalking/blob/e78dd7c2d8d926748f9523e74beb5213ca91386a/test/e2e-v2/cases/apisix/otel-collector/otel-collector-config.yaml, i change oap address and apisix plugin address restart otelcol.
3: i did not found skywalking monitor apisix
it lack any step?

**Reply by wu-sheng** [reply]
You need to check OTEL collector, whether it fetches the metrics from apisix correctly, and forward to skywalking sucessfully.
